hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/compact/ITTestHoodieFlinkCompactor.java

```diff
@@ -253,7 +253,14 @@ public void testHoodieFlinkCompactorWithPlanSelectStrategy(boolean enableChangel
     // wait for the asynchronous commit to finish
     TimeUnit.SECONDS.sleep(3);
 
-    compactionInstantTimeList.add(scheduleCompactionPlan(metaClient, writeClient));
+    metaClient.reloadActiveTimeline();
+    Option<String> compactionInstantTimeOption = CompactionUtil.getCompactionInstantTime(metaClient);
+    // this scheduling is flaky
+    if (compactionInstantTimeOption.isPresent()) {
```

Comment:
Should we do something like this for `testHoodieFlinkCompactorService` as well? That's also flaky for me locally.